### PR TITLE
Add crud test for nested objects

### LINKED ISSUE #375: Add crud tests for nested objects
### I'm submitting a 
- [ ] bug report.
- [x] feature request.

### Current Behaviour:

Currently, there are no tests in test_crud.py to test crud operations on nested objects like 
```json
{"@type": "Command", "DroneID": "1223", 
"State": {"@type": "State", "DroneID": "-1000", "Battery": "50", 
"Direction": "North", "Position": "1,1", "Status": "Active", "Speed": "100"}
} 
```

### Expected Behaviour:

Tests should be there.

### Steps to reproduce:

See test_crud.py in `hydrus/tests`
### Do you want to work on this issue?

NO

### PR DESCRIPTION
**Pull request by vddesai1871**
Fixes #375 

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.5.2 and above.

### Change logs
Added test for nested objects
Adapt tests to use UUIDs
Corrected mypy type annotation in crud.py

### CODE REVIEW

File: hydrus/tests/test_crud.py [hunk 1]
```diff
@@ -194,11 +214,13 @@ def test_insert_type(self):
...
         assert 400 == response_code
 
-    def test_insert_id(self):
+    def test_insert_used_id(self):
         """Test CRUD insert when used ID is given."""
         object_ = gen_dummy_object(random.choice(
             self.doc_collection_classes), self.doc)
-        id_ = "1"
+        id_ = str(uuid.uuid4())
+        insert_response = crud.insert(
```
Comment (chrizandr) [inline hydrus/tests/test_crud.py]: Why this insert out of the try catch block?

Comment (vddesai1871) [reply hydrus/tests/test_crud.py]: The first insert is a normal insert with a unique ID and the other one in the try block is redundant (using already used id, which is supposed to fail).

Comment (vddesai1871) [reply hydrus/tests/test_crud.py]: I have also changed the name of this test to make it more clear.

**Comment by xadahiya** [comment]
@chrizandr is it okay to merge?

**Pull request merged**